tests/foreman/api/test_provisioning.py

```diff
@@ -187,126 +186,142 @@ def test_verify_bugzilla_1114640(self, target_sat):
+            description=new_desc,
+            minor=new_minor_version,
+            major=new_major_version,
+            family=new_os_family,
+            architecture=[new_arch],
+            password_hash=new_pass_hash,
+            provisioning_template=[new_template],
+            ptable=[new_ptable],
+            medium=[new_medium],
+        ).update(
+            [
+                'name',
+                'description',
+                'minor',
+                'major',
+                'family',
+                'architecture',
+                'password_hash',
+                'provisioning_template',
+                'ptable',
+                'medium',
+            ]
+        )
+        assert os.name == new_name
+        assert os.family == new_os_family
+        assert os.minor == new_minor_version
+        assert os.major == new_major_version
+        assert os.description == new_desc
+        assert os.architecture[0].id == new_arch.id
+        assert os.password_hash == new_pass_hash
+        assert os.ptable[0].id == new_ptable.id
+        assert os.provisioning_template[0].id == new_template.id
+        assert os.medium[0].id == new_medium.id
+        # Delete OS
+        module_target_sat.api.OperatingSystem(id=os.id).delete()
+        with pytest.raises(HTTPError):
+            module_target_sat.api.OperatingSystem(id=os.id).read()
 
     @pytest.mark.tier1
-    @pytest.mark.parametrize('pass_hash', **parametrized(('SHA256', 'SHA512')))
-    def test_positive_create_with_password_hash(self, pass_hash, target_sat):
-        """Create operating system with valid password hash option
+    @pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+    def test_positive_create_with_name(self, name, target_sat):
+        """Create operating system with valid name only
 
-        :id: 00830e71-b414-41ab-bc8f-03fd2fbd5a84
+        :id: e95707bf-3344-4d85-866f-4642a8f66cff
 
         :parametrized: yes
 
         :expectedresults: Operating system entity is created and has proper
-            password hash type
+            name
 
         :CaseImportance: Critical
         """
-        os = target_sat.api.OperatingSystem(password_hash=pass_hash).create()
-        assert os.password_hash == pass_hash
-
-    @pytest.mark.tier2
-    def test_positive_create_with_arch(self, target_sat):
-        """Create an operating system that points at an architecture.
-
-        :id: 6a3f7183-b0bf-4834-8c69-a49fe8d7ee5a
-
-        :expectedresults: The operating system is created and points at the
-            given architecture.
-
-        :CaseLevel: Integration
-        """
-        arch = target_sat.api.Architecture().create()
-        operating_sys = target_sat.api.OperatingSystem(architecture=[arch]).create()
-        assert len(operating_sys.architecture) == 1
-        assert operating_sys.architecture[0].id == arch.id
+        os = target_sat.api.OperatingSystem(name=name).create()
+        assert os.name == name
```

Comment:
Can we add finalizer step here for the teardown of deleting OS cretated and in other tests creating OS?

Reply:
We can but the OS is created in different orgs which won't affect other OS tests. same we followed in OS cli tests and we do have deleting OS scenario covered.